scripts/envs/open_manipulator/open_manipulator_reacher_env.py

```diff
@@ -0,0 +1,268 @@
+        """
+        if action is None:
+            action = np.array([1, 1, 1, 1, 1, 1])
+
+        self.prev_tic = self.tic
+        self.tic = time.time()
+        self.elapsed = time.time() - self.prev_tic
+        self.done = False
+
+        if self.step_cnt == self.max_steps:
+            self.done = True
+            self.step_cnt = 0
+
+        act = action.flatten().tolist()
+        self.ros_interface.set_joints_position(act)
+        curr_dist = self._get_dist()
+
+        if not self.is_real:
+            self.reward = self._compute_reward()
+            if self._check_for_termination():
+                print("======================================================")
+                print("Terminates current Episode : OUT OF BOUNDARY")
+                print("======================================================")
+            elif self._check_for_success():
+                print("======================================================")
+                print("Succeeded current Episode")
+                print("======================================================")
+        _joint_pos, _joint_vels, _joint_effos = self.ros_interface.get_joints_states()
+
+        if np.mod(self.step_cnt, 10) == 0:
+            if not self.is_real:
+                print("DISTANCE : ", curr_dist)
+            print("PER STEP ELAPSED : ", self.elapsed)
+            print("SPARSE REWARD : ", self.reward_rescale * self.reward)
+            print("CURRNET EE POSITINO: ", self.ros_interface.gripper_position)
+            print("ACIONS: ", act)
+
+        obs = np.array([_joint_pos, _joint_vels, _joint_effos])
+        self.step_cnt += 1
+
+        return obs, self.reward_rescale * self.reward, self.done
+
+    def reset(self):
+        """Attempt to reset the simulator.
+
+        Since we randomize initial conditions, it is possible to get into
+        a state with numerical issues (e.g. due to penetration or
+        Gimbel lock) or we may not achieve an initial condition (e.g. an
+        object is within the hand).
+
+        In this case, we just keep randomizing until we eventually achieve
+        a valid initial
+        configuration.
+
+        Returns:
+            obs (array) : Array of joint position, joint velocity, joint effort
+        """
+        self.ros_interface._reset_gazebo_world()
+        _joint_pos, _joint_vels, _joint_effos = self.ros_interface.get_joints_states()
+        obs = np.array([_joint_pos, _joint_vels, _joint_effos])
+
+        return obs
+
+    def close(self):
+        """Close by rospy shutdown."""
+        rospy.signal_shutdown("done")
+
+    def _check_robot_moving(self):
+        """Check if robot has reached its initial pose.
+
+        Returns:
+            True if not stopped.
+        """
+        while not rospy.is_shutdown():
+            if self.moving_state == "STOPPED":
+                break
+        return True
+
+    def _check_for_success(self):
```

Comment:
I think both 'check for success' and' check for termination' method should be moved to ros_interface class in the future. This should be done after I finish testing the environment.